base/reduce.jl

```diff
@@ -296,46 +296,95 @@ function extrema(itr)
     return (vmin, vmax)
 end
 
-## all & any
+## any
 
-function mapfoldl(f, ::AndFun, itr)
+any(itr)    = any(IdFun(), itr)
+any(f, itr) = any(specialized_unary(f), itr)
+
+function any(f::IdFun, itr)
     for x in itr
-        !f(x) && return false
+        x && return true
     end
-    return true
+    return false
 end
```

Comment:
Are you sure you need this method? The method below should generate identical code if `f` is `IdFun`.